### ISSUE TITLE
Logging message templates are not making best use of semantic logging practices

### ISSUE THREAD

**Issue by drub0y** [microsoft/AuthJanitor#16]
**TL;DR;** - semantic logging placeholder naming should be treated as any other schema design

----

I've scanned the codebase and from what I can see all logger message templates are using an ordinal based approach for placeholder names. Placeholder names are meant to have semantic value so that, as the values are recorded, they are recorded with a name that the values can later be efficiently reported on/searched by in log aggregation tools.

For example:

```CSharp
Logger.LogInformation("Committing new Secret with name '{0}'", newSecret.Name);
```

Should really be:

```CSharp
Logger.LogInformation("Committing new Secret with name '{SecretName}'", newSecret.Name);
```

If the current, ordinal based placeholder approach is used, values would need to be queried by a field called `"0"` vs the proposed approach where you could actually query by a field named `"SecretName"`. 

Now, the above example is trivial because there's only a single value, but imagine a scenario where you are logging three integer values in the same template and they're logged as `"0"`, `"1"` and `"2"` now. As the person who's trying to write a query to identify every time the value of "remaining inventory" has crossed below the value `100`, how can I easily tell which of those values I'm supposed to query against? I'd have to actually go look at the message template and see where the value is and then reverse engineer it. What if instead the message template had used the placeholder names more like `"InventoryUsed"`, `"RemainingInventory"`, `"NumberOfRubberChickensHarmedDuringInventoryRetrieval"`? Well, now it's pretty clear which one I'm interested in. 

Next, keep in mind these names actually end up in aggregate queries too, so the person who first writes the query might go figure out that the number they're interested in was in field `"1"`, but imagine opening a sufficiently complex aggregate query up later and trying to make heads or tails of it if everything is named "0", "1" and "2" across hundreds of log message templates.

Finally, imagine someone later comes in and edits the message to add a new piece of data, but they kinda re-write the message a bit and put things in a different order. If the names were ordinal, there's a good chance someone tries to reorder the values so that they increase naturally. If they did this, all queries that worked for the original version of the message would start breaking because the values would be offset from their original ordinal names. This would not happen if meaningful placeholder names were used instead.
